src/config.rs

```diff
@@ -362,6 +364,16 @@ impl Cfg {
         Toolchain::from(self, name)
     }
 
+    pub fn get_toolchains_from_glob(
+        &self,
+        pattern: Pattern,
```

Comment:
Is it right for this to take `Pattern` rather than `&Pattern` ?  I realise in this case it's not massively relevant but as a general API design I feel like `&Pattern` might be right.

Reply:
We could theoretically take `&Pattern`, but then we run into lifetime issues, so I didn't think it was worth it given that it will make the code more complex. I am still open to changing it though.

Reply:
I see, okay, no problem